### PR TITLE
Add default security headers

### LINKED ISSUE #267: Add security headers
The website is missing most of the security headers: https://securityheaders.com/?q=jsonforms.io&hide=on&followRedirects=on

They can probably be added similar to this PR: https://github.com/eclipse-glsp/glsp-website-source/pull/86/files

### PR DESCRIPTION
**Pull request by TheZoker**
Fixes #267 

https://securityheaders.com/?q=https%3A%2F%2Fdeploy-preview-271--jsonforms.netlify.app%2F&hide=on&followRedirects=on

### CODE REVIEW

**Review by sdirix — CHANGES_REQUESTED** [review]
If I understand Netlify's docs correctly then you need to create/modify a `_headers` file in the `build` directory. The toml is static and will not be read dynamically for each deployment.

Although it seems to work for the preview.....

**Comment by TheZoker** [comment]
Hmm according to this doc, it does not seem to make a difference if we put it into the `_headers` file or config file:
https://docs.netlify.com/routing/headers/#syntax-for-the-netlify-configuration-file

> You can configure custom headers for your Netlify site in two ways:

> - Save a plain text file called _headers to the [publish directory](https://docs.netlify.com/configure-builds/overview/#basic-build-settings) of your site. You can find [_headers file syntax](https://docs.netlify.com/routing/headers/#syntax-for-the-headers-file) details below.
> - Add one or more headers tables to your [Netlify configuration file](https://docs.netlify.com/configure-builds/file-based-configuration/#headers). This method allows for more structured configuration and additional capabilities, as described in the [Netlify configuration file syntax](https://docs.netlify.com/routing/headers/#syntax-for-the-netlify-configuration-file) section below.


This uses also the same method as in the linked [PR](https://github.com/eclipse-glsp/glsp-website-source/pull/86/files)

File: netlify.toml [hunk 1]
```diff
@@ -28,3 +28,13 @@
...
  from = "/public/api/vue2/*"
  to = "/api/vue2/"
+
+[[headers]]
+   for = "/*"
+   [headers.values]
+      Access-Control-Allow-Origin = "*"
+      Referrer-Policy = "strict-origin-when-cross-origin"
+      Content-Security-Policy = "default-src 'self'; script-src 'self' 'unsafe-eval' 'sha256-mC5lwOEBZZZXJoN3sDvzxnxAdNIEKujq9NSXgmhc4HM=' 'sha256-iIf+c/EMxKD/FXoUDy0YsZ3mE+JhzPsmR+aVbrjkdwM=' static.cloudflareinsights.com; style-src 'self' 'unsafe-inline'; img-src 'self' data:; object-src 'none'; frame-ancestors 'self';"
+      X-Frame-Options = "SAMEORIGIN"
```
Comment (sdirix) [inline netlify.toml]: Shouldn't there be a placeholder here?

Comment (TheZoker) [reply netlify.toml]: Yes, but since the calculated hash was always wrong, I inserted the expected hash directly.

File: build.sh [hunk 2]
```diff
@@ -20,3 +24,16 @@ if [[ $WITH_NEXT = "next" ]]; then
...
+#
+
+# Download the file
+curl -O https://static.cloudflareinsights.com/beacon.min.js
+
+# Calculate the sha256 hash of the file and convert it to base64
+hash=$(openssl dgst -sha256 -binary beacon.min.js | openssl base64)
+
+# Replace the placeholder in netlify.toml file with the calculated hash
+sed -i "s|##HASH_PLACEHOLDER##|'sha256-$hash'|g" netlify.toml
```
Comment (sdirix) [inline build.sh]: This script needs to be adapted so it works with the correct / previous hashs already inserted.

**Comment by sdirix** [comment]
For me the calculated hash looks correct. It's the same hash I get with an online hash calculator. They also recently changed their file again, so the hash might is different now. Can you recheck? With the script you posted I get `wjXyEBe8wR/Kox19/ZhVquvL9fbX7pv58umKkQkHw5E=` which seems correct to me.

**Review by sdirix — APPROVED** [review]
LGTM

**Pull request merged**